### PR TITLE
Patch article by

### PR DESCRIPTION
**Pull request by akin-fagbohun**
new PATCH functionality added to patch articles by ID.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -3,7 +3,8 @@ const request = require('supertest');
 const db = require('../db/connection');
 const express = require('express');
 const seed = require('../db/seeds/seed');
-const testData = require('../db/data/test-data/index')
+const testData = require('../db/data/test-data/index');
+const res = require('express/lib/response');
```
Comment (cavalierswain) [inline __tests__/app.test.js]: be careful of the auto import here!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -55,31 +56,82 @@ describe('ERROR GET /api/topicd', () => {
...
-      expect(output).toHaveProperty('created_at');
-      expect(output).toHaveProperty('votes');
-    });   
+      .get('/api/articles/1')
+      .expect(200)
+      .then((res) => {
+        const output = res.body.rows[0];
+        expect(output).toHaveProperty('author');
+        expect(output).toHaveProperty('title');
+        expect(output).toHaveProperty('article_id');
```
Comment (cavalierswain) [inline __tests__/app.test.js:75]: if we are checking for a specific article then we would want to be a bit more specific with the testing. 
what do we expect the article_id to be? etc.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -55,31 +56,82 @@ describe('ERROR GET /api/topicd', () => {
...
+describe('PATCH /api/articles/:article_id', () => {
+  test('sends PATCH to ID endpoint -> checks each object key', () => {
+    return request(app)
+      .patch('/api/articles/1')
+      .send({ inc_votes : 1 })
+      .expect(201)
+      .then((res) => {
+        const output = res.body.rows[0];
+        expect(output).toBeInstanceOf(Object);
+        expect(Object.keys(output).length).toBe(7);
```
Comment (cavalierswain) [inline __tests__/app.test.js:93]: if we are sending 1 vote, what is the total vote count for the article after is been sent?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -55,31 +56,82 @@ describe('ERROR GET /api/topicd', () => {
...
+      .expect(201)
+      .then((res) => {
+        const output = res.body.rows[0];
+        expect(output).toHaveProperty('author');
+        expect(output).toHaveProperty('title');
+        expect(output).toHaveProperty('article_id');
+        expect(output).toHaveProperty('body');
+        expect(output).toHaveProperty('topic');
+        expect(output).toHaveProperty('created_at');
+        expect(output).toHaveProperty('votes');
```
Comment (cavalierswain) [inline __tests__/app.test.js:109]: here we can be specific with the number of votes that we are getting back

File: __tests__/app.test.js (hunk 3 above)
Comment (akin-fagbohun) [reply __tests__/app.test.js:93]: please see line 120. Also 130 which tests a negative vote's value. Just I opted to place these in separate tests because I recall it being said that we should be careful when making many assertions in one test.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -44,13 +44,142 @@ describe('GET /api/topics', () => {
       });
   });
 });
-describe('GET /api/topisc', () => {
+describe('ERROR GET /api/topicd', () => {
```
Comment (dentednerd) [inline __tests__/app.test.js:47]: Line 47 doesn't match line 50... I'm being super picky here but it's good to keep the details straight!

File: __tests__/app.test.js [hunk 6]
```diff
@@ -44,13 +44,142 @@ describe('GET /api/topics', () => {
...
   });
 });
 
+describe('GET /api/articles/:article_id', () => {
+  test('sends GET to ID endpoint -> returns an object', () => {
+    return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then((res) => {
+        const output = res.body.rows[0];
```
Comment (dentednerd) [inline __tests__/app.test.js]: Your result should already be a single object here, instead of an array that you're having to use [0] on. Please revisit your model.

File: __tests__/app.test.js [hunk 7]
```diff
@@ -44,13 +44,142 @@ describe('GET /api/topics', () => {
...
+        expect(Object.keys(output).length).toBe(7);
+    });   
+  });
+
+  test('sends GET to ID endpoint -> checks each object key', () => {
+    return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then((res) => {
+        const output = res.body.rows[0];
```
Comment (dentednerd) [inline __tests__/app.test.js]: Once again, please revisit your model.

File: models/models.js [hunk 8]
```diff
@@ -1,15 +1,17 @@
...
-//     const results = await db.query('SELECT * FROM topics')
-//     res.send(results.rows)
-//   } catch (err) {
-//     console.log(err);
-//   }
-// };
\ No newline at end of file
+
+exports.selectArticleById = (id) => {
+  return db.query('SELECT * FROM articles WHERE article_id = $1', [id])
```
Comment (dentednerd) [inline models/models.js:11]: What we want to return here is a single article object. Right now you're returning an array of objects.
